packages/ra-core/src/dataProvider/useUpdate.ts

```diff
@@ -91,13 +92,21 @@ export const useUpdate = <RecordType extends RaRecord = any, ErrorType = Error>(
     const dataProvider = useDataProvider();
     const queryClient = useQueryClient();
     const { id, data, meta } = params;
-    const { mutationMode = 'pessimistic', ...mutationOptions } = options;
+    const {
+        mutationMode = 'pessimistic',
+        getMutateMiddlewares,
+        ...mutationOptions
+    } = options;
     const mode = useRef<MutationMode>(mutationMode);
     const paramsRef = useRef<Partial<UpdateParams<RecordType>>>(params);
     const snapshot = useRef<Snapshot>([]);
-    const hasCallTimeOnError = useRef(false);
+    const callTimeOnError = useRef<
+        UseUpdateOptions<RecordType, ErrorType>['onError']
+    >();
     const hasCallTimeOnSuccess = useRef(false);
```

Comment:
By reading it, I'm surprised you don't need to do the same for `onSuccess` as you did for `onError` and `onSettled`. The reason becomes clearer later in the file, but you should explain the difference here.